generate_tidal_profile.py

```diff
@@ -0,0 +1,438 @@
+                tidal,
+                self.latitude,
+                self.longitude,
+                self.advanced_inputs['rated_power'],
+                self.advanced_inputs['rotor_length'],
+                self.advanced_inputs['turbine_number'],
+                self.advanced_inputs['inverter_efficiency'],
+                self.advanced_inputs['maximum_cp'],
+                self.advanced_inputs['turbine_losses'],
+                self.advanced_inputs['cut_in_velocity'],
+                self.advanced_inputs['cut_out_velocity'])]
+
+
+    def crop_timeline(self, num_seconds, validate=True):
+        """ Used to crop the profiles to a period of less than 1 day
+        ## Is this necessary? I kept this the same from 'generate_solar_profile'
+        
+            num_seconds is the number of seconds of the new outage
+                period
+        """
+
+        # Validate parameters
+        if validate:
+            args_dict = {'num_seconds': num_seconds}
+            validate_all_parameters(args_dict)
+
+        # For each profile in tidal_profiles, power_profiles, temp_profiles and
+        #   night_profiles, crop to the specified number of seconds, rounding down to the
+        #   nearest timestep
+        for i in range(len(self.tidal_profiles)):
+            self.tidal_profiles[i] = \
+                self.tidal_profiles[i].loc[
+                :self.tidal_profiles[i].index[0] + datetime.timedelta(
+                    seconds=num_seconds - 1)]
+
+        for i in range(len(self.power_profiles)):
+            self.power_profiles[i] = \
+                self.power_profiles[i].loc[
+                :self.power_profiles[i].index[0] + datetime.timedelta(
+                    seconds=num_seconds - 1)]
+
+
+    def tidal_checks(self):
+        """ Several checks to  make sure the tidal profiles look OK. """
+
+        ## Do we need this? I left this as-is from the solar.
+        # It won't be as obvious as solar if the tidal profile is off, but
+        # users should still be able to look at the profile and see the
+        # tidal peaks for each day
+
+        # Get the profiles with the min and max energy
+        total_energy = [prof.sum() for prof in self.power_profiles]
+        max_profile_num = np.where(total_energy == max(total_energy))[0][0]
+        min_profile_num = np.where(total_energy == min(total_energy))[0][0]
+
+        # Plot the profiles with min and max energy
+        fig = plt.figure()
+        ax1 = fig.add_subplot(121)
+        self.power_profiles[max_profile_num].plot(
+            ax=ax1, title='Profile with max energy generation')
+        ax1.set_ylabel('Power (kW)')
+        ax2 = fig.add_subplot(122)
+        self.power_profiles[min_profile_num].plot(
+            ax=ax2, title='Profile with min energy generation')
+        ax2.set_ylabel('Power (kW)')
+
+    def get_dc_to_ac(self):
+        """ Returns the dc to ac ratio. """
+
+        # Should we keep this? I don't think this matters
```

Comment:
We probably don't need this, esp since we don't have the 'inverter_output' param defined.